torch/onnx/symbolic_opset11.py

```diff
@@ -475,14 +475,21 @@ def size(g, self, dim=None):
 
 def squeeze(g, self, dim=None):
     if dim is None:
-        dims = []
-        for i, size in enumerate(self.type().sizes()):
-            if size == 1:
-                dims.append(i)
-    else:
-        dims = [sym_help._get_const(dim, 'i', 'dim')]
-    return g.op("Squeeze", self, axes_i=dims)
-
+        return g.op("Squeeze", self)
+
+    dim = sym_help._get_const(dim, 'i', 'dim')
+
+    # create 'cond' node (condition is shape[i]==1)
+    dim_constant = g.op("Constant", value_t=torch.tensor([dim]))
```

Comment:
Can we directly check the self's size instead of inserting IF-else block?

Reply:
The size is not always available at export time, so we needed to add an if-else block.